doc/kci_rootfs.md

```diff
@@ -0,0 +1,156 @@
+    ```
+5. To list available rootfs configuration, we can use `list_configs` option.
+
+    ```
+    $ ./kci_rootfs list_configs       
+
+    buster
+
+    buster-v4l2
+
+    buster-igt
+    ```
+   By default `kci_rootfs` reads entries from `rootfs-configs.yaml`. This file acts as a rootfs config file.
+
+6. Lets list all available rootfs images using `./kci_rootfs list_variants`. It should show existing
+rootfs name along with its architecture. 
+
+    ```
+    buster amd64
+    buster arm64
+    ```
+
+    It also comes with couple of options `--config` and `--arch` to filter the results based on config name or arch type.
+    
+    ```
+    $ ./kci_rootfs list_variants --config buster --arch i386
+    buster i386
+
+    $ ./kci_rootfs list_variants --config buster 
+    buster amd64
+    buster arm64
+
+    $ ./kci_rootfs list_variants --arch amd64
+    buster amd64
+    buster-v4l2 amd64
+    buster-igt amd64
+    ```
+7. Now its time to re-build existing rootfs image using `kci_rootfs build`. It takes three arguments,
+`--config` refers to rootfs name which we want to build. `--data-path` points to debos files location.
+`--arch` refers to CPU arch we want to build. For example, to build a buster rootfs image for i386, 
+we can run
+
+    ```
+    ./kci_rootfs build --config buster --data-path jenkins/debian/debos --arch i386
+
+    ```
+   depending on your network speed, this will take sometime to complete. 
+
+8. If build is successful you should see message like
+
+    ```
+    cd ${ROOTDIR} ; find -H  |  cpio -H newc -v -o | gzip -c - > ${ARTIFACTDIR}/buster/i386/rootfs.cpio.gz | ./build_info.json
+    cd ${ROOTDIR} ; find -H  |  cpio -H newc -v -o | gzip -c - > ${ARTIFACTDIR}/buster/i386/rootfs.cpio.gz | 79539 blocks
+    Powering off.
+    ==== Recipe done ====
+    ```
+    Finally newly built rootfs images can be found at `jenkins/debian/debos/buster/i386/`
+
+    ```
+    $ ls jenkins/debian/debos/buster/i386/
+    build_info.json  full.rootfs.cpio.gz  full.rootfs.tar.xz  initrd.cpio.gz  rootfs.cpio.gz  rootfs.ext4.xz
+    ```
+
+### Create a new rootfs image 
+
+Now we know how to build default kci_rootfs images. Lets look at how to add simple `stretch` rootfs image. 
+
+1. First we need to add appropriate entries to `rootfs_config.yml` file.
+
+    ```
+      stretch:
+        rootfs_type: debos
+        debian_release: stretch
+        arch_list:
+          - amd64
+          - arm64
+        extra_packages_remove:
+          - e2fslibs
+          - e2fsprogs
+    ```
+
+  Above entry will create rootfs named `stretch` for CPU amd64 and arm64 architectures without `e2fslibs` and  `e2fsprogs` packages. List of possible `rootfs-config` yaml entries and its description are listed below:
+
+  | Entry                 | Description |
+  | ----------------------| ----------- |
+  | stretch               | Unique rootfs configuration name. |
+  | rootfs_type           | Tool used for rootfs creation. |
```

Comment:
Maybe it's worth mentioning that at the moment the only acceptable value of `rootfs_type` is `debos` as well as the options below are debos specific.